corehq/apps/userreports/expressions/specs.py

```diff
@@ -79,11 +83,12 @@ def configure(self, context):
             raise BadSpecError(u'Name {} not found in list of named expressions!'.format(self.name))
         self._context = context
 
-    def _context_cache_key(self):
-        return 'named_expression-{}'.format(self.name)
+    def _context_cache_key(self, item):
+        item_hash = hashlib.md5(json.dumps(item, cls=DjangoJSONEncoder, sort_keys=True)).hexdigest()
```

Comment:
@emord why is it necessary to hash the full object here. Can we not just use the ID? for cases where the item is a repeat or something like that it may not have an ID so the we could use the hash. Also wondering if we can't use the object ID `id(item)` rather than generating a hash?

Reason for asking is that this eagerly evaluates any lazy fields in the item.

Reply:
That seems reasonable to me. Its likely not necessary to hash the whole object, just what I thought to do at the time. as long as tests pass with `id` i'm happy with the change